compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -700,44 +700,43 @@ execMigrate projectOpts opts0 inFile1_ inFile2_ mbDir = do
+            putStrLn $ "created workdir: " <> workDir
+            withCurrentDirectory workDir $ do
+                unless
                 -- TODO (drsk) remove this filter
-                (iuid `elem`
-                 map stringToInstalledUnitId ["daml-prim", "daml-stdlib"]) $ do
+                    (iuid `elem` map stringToInstalledUnitId ["daml-prim"]) $
                   -- typecheck and generate interface files
-                  forM_ src $ \(fp, content) -> do
-                      let path = fromNormalizedFilePath fp
-                      createDirectoryIfMissing True $ takeDirectory path
-                      writeFile path content
-                  opts' <-
-                      mkOptions $
-                      opts
-                          { optWriteInterface = True
-                          , optPackageDbs = [projectPkgDb]
-                          , optIfaceDir =
-                                Just
-                                    (dbPath </> installedUnitIdString iuid)
-                          , optIsGenerated = True
-                          , optMbPackageName =
-                                Just $ installedUnitIdString iuid
-                          }
-                  withDamlIdeState opts' loggerH diagnosticsLogger $ \ide ->
-                      forM_ src $ \(fp, _content) -> do
-                          mbCore <-
-                              runAction ide $
-                              getGhcCore fp
-                          case mbCore of
-                              Nothing ->
-                                  ioError $
-                                  userError $
-                                  "Compilation of generated source for " <>
-                                  installedUnitIdString iuid <>
-                                  " failed."
-                              Just _core -> pure ()
+                 do
+                    forM_ src $ \(fp, content) -> do
+                        let path = fromNormalizedFilePath fp
+                        createDirectoryIfMissing True $ takeDirectory path
+                        writeFile path content
+                    opts' <-
+                        mkOptions $
+                        opts
+                            { optWriteInterface = True
+                            , optPackageDbs = [projectPkgDb]
+                            , optIfaceDir = Just (dbPath </> installedUnitIdString iuid)
+                            , optIsGenerated = True
+                            , optMbPackageName = Just $ installedUnitIdString iuid
+                            }
+                    withDamlIdeState opts' loggerH diagnosticsLogger $ \ide ->
+                        forM_ src $ \(fp, _content) -> do
+                            mbCore <- runAction ide $ getGhcCore fp
+                            case mbCore of
+                                Nothing ->
+                                    ioError $
```

Comment:
You can use `fail` instead of `ioError . userError`

Reply:
ah nice, there's many places in damlc where we could change this.